### PR TITLE
Add Arm Subsystem

### PR DESCRIPTION
**Pull request by JashKotadiya**

### CODE REVIEW

**Comment by AdamCraftmaster** [comment]
Needs to pull main

File: .vscode/settings.json [hunk 1]
```diff
@@ -45,5 +45,8 @@
   },
   "wpilib.deployOffline": true,
   "wpilib.offline": false,
-  "wpilib.skipTests": false
+  "wpilib.skipTests": false,
+  "[java]": {
+    "editor.defaultFormatter": "mwpb.java-prettier-formatter"
```
Comment (AdamCraftmaster) [inline .vscode/settings.json:50]: This should not be here, this is for individuals only

File: src/main/java/frc/robot/commands/arm/ArmStopCMD.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  public void execute() {
+    arm.stop();
+  }
+
+  @Override
+  public void end(boolean interrupted) {}
+
+  @Override
+  public boolean isFinished() {
+    return false; // TODO: arm.isStopped();
```
Comment (AdamCraftmaster) [inline src/main/java/frc/robot/commands/arm/ArmStopCMD.java:28]: Fix this TODO

File: src/main/java/frc/robot/constants/CompBotConstants.java [hunk 3]
```diff
@@ -125,4 +125,65 @@ public FeedbackConstant chassisDriveFB() {
...
+
+  @Override
+  public double armExtendMin() {
+    return 0.02;
+  }
+
+  @Override
+  public double armRotateMax() {
+
+    return 180;
```
Comment (AdamCraftmaster) [inline src/main/java/frc/robot/constants/CompBotConstants.java:177]: Is this in degrees or radians?

File: src/main/java/frc/robot/constants/RobotArmConstants.java [hunk 4]
```diff
@@ -0,0 +1,185 @@
+package frc.robot.constants;
```
Comment (AdamCraftmaster) [inline src/main/java/frc/robot/constants/RobotArmConstants.java:1]: This file should be deleted

File: src/main/java/frc/robot/subsystems/arm/Arm.java [hunk 5]
```diff
@@ -0,0 +1,275 @@
...
+          mode = ArmMode.CALIBRATE;
+          extendCalibrated = false;
+        } else {
+          hold();
+        }
+      }
+    }
+    armIO.updateInputs(armIOInputs);
+    logger.processInputs("Arm", armIOInputs);
+    logger.recordOutput("Arm/mode", mode.toString());
```
Comment (AdamCraftmaster) [inline src/main/java/frc/robot/subsystems/arm/Arm.java:117]: For this and all other outputs, it should be in UpperCamel case

File: src/main/java/frc/robot/subsystems/arm/ArmIOSim.java [hunk 6]
```diff
@@ -0,0 +1,53 @@
+package frc.robot.subsystems.arm;
```
Comment (AdamCraftmaster) [inline src/main/java/frc/robot/subsystems/arm/ArmIOSim.java:1]: This file should be rewritten or deleted

File: src/main/java/frc/robot/subsystems/conevision/ConeTargetDistance.java [hunk 7]
```diff
@@ -0,0 +1,3 @@
+package frc.robot.subsystems.conevision;
```
Comment (AdamCraftmaster) [inline src/main/java/frc/robot/subsystems/conevision/ConeTargetDistance.java:1]: This file should be deleted

File: src/main/java/frc/robot/Robot.java [hunk 8]
```diff
@@ -35,18 +35,6 @@ public class Robot extends LoggedRobot {
   public void robotInit() {
     Logger logger = Logger.getInstance();
 
-    // Record metadata
```
Comment (AdamCraftmaster) [inline src/main/java/frc/robot/Robot.java:38]: Do not delete metadata, add it back

File: src/main/java/frc/robot/Robot.java [hunk 9]
```diff
@@ -57,7 +45,7 @@ public void robotInit() {
           logger.addDataReceiver(new WPILOGWriter(folder));
         }
         if (RobotConstants.get().robot() == RobotType.ROBOT_COMP) {
-          new PowerDistribution(63, ModuleType.kRev);
+          //   new PowerDistribution(63, ModuleType.kRev);
```
Comment (AdamCraftmaster) [inline src/main/java/frc/robot/Robot.java:48]: This should be uncommented

File: src/main/java/frc/robot/RobotConstants.java [hunk 10]
```diff
@@ -31,6 +32,9 @@ private static void initConstants() throws IOException {
           case "von neumann":
             RobotConstants.set(new CompBotConstants());
             break;
+          case "hopper":
```
Comment (AdamCraftmaster) [inline src/main/java/frc/robot/RobotConstants.java:35]: This case should be removed

File: src/main/java/frc/robot/RobotContainer.java [hunk 11]
```diff
@@ -128,6 +138,14 @@ public RobotContainer() {
       }
     }
 
+    arm =
```
Comment (AdamCraftmaster) [inline src/main/java/frc/robot/RobotContainer.java:141]: This should be moved into the cases, in order to ensure that sim ignores it (will fix this in a commit soon)

**Review by AdamCraftmaster — CHANGES_REQUESTED** [review]
A few things to look at

File: src/main/java/frc/robot/constants/CompBotConstants.java (hunk 3 above)
Comment (elleryjh) [reply src/main/java/frc/robot/constants/CompBotConstants.java]: Renamed to indicate meters.

**Review by AdamCraftmaster — APPROVED** [review]
lgtm

**Pull request merged**